src/storage/src/hummock/compactor/mod.rs

```diff
@@ -619,15 +676,20 @@ impl Compactor {
                                                     "Cancellation of compaction task failed. task_id: {}",
                                                     cancel_compact_task.task_id
                                                 );
+                                            }
+                                        } else {
+                                            tracing::warn!(
+                                                    "Attempting to cancel non-existent compaction task. task_id: {}",
+                                                    cancel_compact_task.task_id
+                                                );
                                         }
-                                    } else {
-                                        tracing::warn!(
-                                                "Attempting to cancel non-existent compaction task. task_id: {}",
-                                                cancel_compact_task.task_id
-                                            );
+                                    }
+
+                                    ResponseEvent::PullTaskAck(_pull_task_ack) => {
+                                        // set flag
+                                        pull_task_ack.store(true, Ordering::SeqCst);
```

Comment:
Seems `pull_task_ack` is only accessed within this loop. I think it can just be a regular bool and there is no need to use atomic.

Reply:
This is not really feasible, async move will catch it but we have to share it